src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -164,163 +182,90 @@ function BaseVideoPlayer({
+    useEventListener(videoPlayerRef.current, 'playToEnd', () => {
+        setIsEnded(true);
+        setControlStatusState(CONST.VIDEO_PLAYER.CONTROLS_STATUS.SHOW);
+        controlsOpacity.set(1);
+    });
 
-            if (event.fullscreenUpdate === VideoFullscreenUpdate.PLAYER_DID_DISMISS) {
-                if (videoPlayerElementParentRef.current && 'removeEventListener' in videoPlayerElementParentRef.current) {
-                    videoPlayerElementParentRef.current.removeEventListener('wheel', stopWheelPropagation);
-                }
-                isFullScreenRef.current = false;
-
-                // Sync volume updates in full screen mode after leaving it
-                currentVideoPlayerRef.current?.getStatusAsync?.().then((status) => {
-                    if (!('isMuted' in status)) {
-                        return;
-                    }
-                    updateVolume(status.isMuted ? 0 : status.volume || 1);
-                });
-
-                // we need to use video state ref to check if video is playing, to catch proper state after exiting fullscreen
-                // and also fix a bug with fullscreen mode dismissing when handleFullscreenUpdate function changes
-                if (videoStateRef.current && (!('isPlaying' in videoStateRef.current) || videoStateRef.current.isPlaying)) {
-                    pauseVideo();
-                    playVideo();
-                    videoResumeTryNumberRef.current = 3;
-                }
-            }
-        },
-        [isFullScreenRef, onFullscreenUpdate, pauseVideo, playVideo, videoResumeTryNumberRef, updateVolume, currentVideoPlayerRef, stopWheelPropagation],
-    );
+    useEventListener(videoPlayerRef.current, 'sourceLoad', (event) => {
+        onSourceLoaded?.(event);
+    });
 
-    const bindFunctions = useCallback(() => {
-        const currentVideoPlayer = currentVideoPlayerRef.current;
-        if (!currentVideoPlayer) {
+    useEffect(() => {
+        if (!videoPlayerRef.current.duration) {
             return;
         }
-        currentVideoPlayer._onPlaybackStatusUpdate = handlePlaybackStatusUpdate;
-        currentVideoPlayer._onFullscreenUpdate = handleFullscreenUpdate;
-
-        // update states after binding
-        currentVideoPlayer.getStatusAsync().then((status) => {
-            handlePlaybackStatusUpdate(status);
-        });
-    }, [currentVideoPlayerRef, handleFullscreenUpdate, handlePlaybackStatusUpdate]);
+        setDuration(videoPlayerRef.current.duration);
+    }, [videoPlayerRef.current.duration]);
 
     // use `useLayoutEffect` instead of `useEffect` because ref is null when unmount in `useEffect` hook
     // ref url: https://reactjs.org/blog/2020/08/10/react-v17-rc.html#effect-cleanup-timing
+
+    useEffect(() => {
+        mountedVideoPlayersRef.current.push(url);
+        return () => {
+            const urlIndex = mountedVideoPlayersRef.current.indexOf(url);
+            mountedVideoPlayersRef.current.splice(urlIndex, 1);
```

Comment:
ESLint is failing due to a valid reason. I think extracting the ref value to a variable should help here